packages/csvviewer/src/widget.ts

```diff
@@ -81,12 +81,17 @@ class CSVViewer extends Widget {
     layout.addWidget(this._grid);
 
     context.pathChanged.connect(this._onPathChanged, this);
-    // Throttle the rendering rate of the widget.
-    this._monitor = new ActivityMonitor({
-      signal: context.model.contentChanged,
-      timeout: RENDER_TIMEOUT
+
+    this._context.ready.then(() => {
+      this._updateGrid();
+      this._ready.resolve(undefined);
+      // Throttle the rendering rate of the widget.
+      this._monitor = new ActivityMonitor({
+        signal: context.model.contentChanged,
+        timeout: RENDER_TIMEOUT
+      });
+      this._monitor.activityStopped.connect(this._updateGrid, this);
```

Comment:
What happens when the model changes and the grid is updated? Will the ready promise fire again? Should it?